platforms/posix/src/px4/common/main.cpp

```diff
@@ -662,3 +673,100 @@ int change_directory(const std::string &directory)
 
 	return PX4_OK;
 }
+
+#else
+extern "C" int LLVMFuzzerTestOneInput(const uint8_t *data, const size_t size)
+{
+	register_sig_handler();
+
+	initialize_fake_px4_once();
+
+	send_mavlink(data, size);
+
+	return 0;
+}
+
+void initialize_fake_px4_once()
```

Comment:
Would it make sense to throw this into an entirely separate file and have clear entry points (regular main vs fuzzing)? 

The regular SITL start is already kind of convoluted and has other issues we need to address (clean shutdown, etc).

Reply:
It would, yes :smile:. I'll give it a try.

Reply:
Done, have a look :)